hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/PerformanceEvaluation.java

```diff
@@ -1149,7 +1149,7 @@ private static long nextRandomSeed() {
       return randomSeed.nextLong();
     }
 
-    private final int everyN;
+    private final long everyN;
 
     protected final Random rand = new Random(nextRandomSeed());
```

Comment:
So we can remove this now?

Reply:
Yes, we can. I didn't remove it in the first place because I wanted to minimize the code change, but now that we agree on using ThreadLocalRandom, let me clean this up.

Reply:
On the second thought, we can keep this member and just set it to `ThreadLocalRandom.current()` to minimize the required code change. `TestBase` instances are created on the fly by each thread in the pool, so I think it's safe to do this. What do you think?

```diff
diff --git a/hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/PerformanceEvaluation.java b/hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/PerformanceEvaluation.java
index ffe33a9e8b..7bbccbd48d 100644
--- a/hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/PerformanceEvaluation.java
+++ b/hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/PerformanceEvaluation.java
@@ -1141,17 +1141,9 @@ public class PerformanceEvaluation extends Configured implements Tool {
    * A test. Subclass to particularize what happens per row.
    */
   static abstract class TestBase {
-    // Below is make it so when Tests are all running in the one
-    // jvm, that they each have a differently seeded Random.
-    private static final Random randomSeed = new Random(EnvironmentEdgeManager.currentTime());
-
-    private static long nextRandomSeed() {
-      return randomSeed.nextLong();
-    }
-
     private final long everyN;
 
-    protected final Random rand = new Random(nextRandomSeed());
+    protected final Random rand = ThreadLocalRandom.current();
     protected final Configuration conf;
     protected final TestOptions opts;
 
```

Reply:
> to minimize the required code change

`generateData` is used in other places, but without a custom seed value.

https://github.com/apache/hbase/blob/fd11b9bfcf2074ff0910ca1278f8a2fb66f0ad58/hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/mapreduce/TestCellBasedHFileOutputFormat2.java#L508-L518

https://github.com/apache/hbase/blob/b7def4ff9b98164152ea4ccff6206bfe3d17bac6/hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/mapreduce/TestHFileOutputFormat2.java#L545-L555

So it's actually trivial to remove the Random parameter altogether.

Reply:
So I have pushed two commits.

* [756ca4e](https://github.com/apache/hbase/pull/6559/commits/756ca4e99e95753c2a6b105231b4c547b9617a2c) is a minimal change that only replaces the value of `rand` to `ThreadLocalRandom.current()`
* [09099f7](https://github.com/apache/hbase/pull/6559/commits/09099f7adbf54cff32f5af02263cd2aa0299ab71) then completely removes Random parameters from the methods.

Please let me know which version you prefer.